fact/src/config/builder.rs

```diff
@@ -0,0 +1,59 @@
+use std::{fs::read_to_string, path::PathBuf, sync::LazyLock};
+
+use anyhow::Context;
+use clap::Parser;
+
+use crate::config::FactCli;
+
+use super::FactConfig;
+
+pub(super) struct FactConfigBuilder {
+    files: Vec<PathBuf>,
+}
+
+impl FactConfigBuilder {
+    pub(super) fn new() -> Self {
+        let files = Vec::new();
+        FactConfigBuilder { files }
+    }
+
+    pub(super) fn add_files(
+        mut self,
+        files: &[impl Into<PathBuf> + AsRef<std::ffi::OsStr>],
+    ) -> Self {
+        for file in files {
+            self.files.push(file.into());
+        }
+        self
+    }
+
+    pub(super) fn files(&self) -> &[PathBuf] {
```

Comment:
This is a bit of a nit, so feel free to tell me where to go with this one, but this feels like it's breaking the builder pattern a fair bit. The builder should be a bit more ephemeral (and it might even make sense for `build` to consume the builder) 

But these files (as `PathBuf`) are used extensively later on, long after the builder has been used to build the config.

Perhaps we could expose this from the config itself since it is the one that owns the content of these files and should know which files it got that information from? WDYT?